### ISSUE TITLE
Muting a single channel mutes all channels with pulseaudio installed

### ISSUE THREAD

**Issue by milos3** [nicklan/pnmixer#70]
Using Xubuntu 14.04 on a Intel laptop. Without pulseaudio muting/unmuting a single channel works normal. But when pulseaudio is installed: 
- Sound Device: Card: (default): only Master channel is available which mutes and unmutes all channels, tested using alsamixer. At the end it works fine.
- Sound Device: Card: (HDA Intel): all channels are available here and if I mute a **single** choosen channel, it mutes **all** channels. And when I try to unmute it, it only unmutes that one channel and leaves all  the rest channels muted. To unmute the rest I need to use alsamixer. 

This is suppose pulseaudio bug which happens when muting a single channel through alsa. The bugs can be found at https://bugs.launchpad.net/ubuntu/+source/pulseaudio/+bug/878986 and https://bugs.launchpad.net/ubuntu/+source/alsa-utils/+bug/1026331.

One possible solution would be to mute/ unmute through pulseaudio when it's installed?

**Comment by hasufell** [comment]
There's currently no direct pulseaudio support and we haven't found anyone who is willing to work on it.

**Comment by elboulangero** [comment]
Yep, for me the whole purpose of PNMixer is to have a good sound mixer for systems that don't have pulseaudio.
Maybe a good thing would be to be more explicit about the fact that we don't support pulseaudio, like issue a warning, or refuse to start, or something.

**Comment by hasufell** [comment]
Refuse to start would be quite harsh :)

I'd say we can make the README more explicit about it. But if someone wants to work on pulseaudio support, why not.

**Comment by elboulangero** [comment]
I'm not the guy for this, sorry ;)

**Comment by elboulangero** [comment]
Hi @milos3,

if you wish you can checkout the latest PNMixer version (branch Master) and give it a try. It seems that there have been some improvements with PulseAudio after we rewrote PNMixer, but we need more tests since none of us (developers) use PA. Feedback appreciated !

**Comment by milos3** [comment]
I just tried the new build and it's pretty much the same as before. Is still mutes all channels when muting a single channel with HDA Intel device. But at the end it still works fine with default device.

**Comment by elboulangero** [comment]
OK, thx for the feedback. I don't think there will be any improvement on that matter until someone is really willing to write a pulseaudio backend for pnmixer. But it's always good to have up-to-date feedback, at least we know were we are now. Thx :)

**Comment by raveit65** [comment]
I did run in same issue here in fedora 26. 
Tested it with pavucontrol, mate-volume-control or xfce4-mixer as mixer backend. 
It is possible to mute/unmute the sound card with keyboard (fn+audo-mute) without any problem.
So i think this isn't a general problem here.
I'd like to use pnmixer as default mixer-applet for next fedora release as fedora maintainer, because i like the flexibility to use different mixer backends .
But unmuting should really work.
Can you please take a look into it again, please?
I am happy to be give you any information you need.

**Comment by hasufell** [comment]
> `fn+audo-mute`

I don't know what that means. That could run any kind of code/command, depending on your setup.

**Comment by raveit65** [comment]
> I don't know what that means. That could run any kind of code/command, depending on your setup.
This is a standard keyboard command fn+audiomute
Don't know if this is the right name, but every notebook keyboard have this.

**Comment by hasufell** [comment]
I am pretty sure it does nothing, except when your DE (or wm, or xbindkeys, or whatever) is set up to do something with it. So unless I know what code gets actually triggered for you, I cannot comment on that.

**Comment by raveit65** [comment]
Which command should i use to mute/unmute my card?

**Comment by hasufell** [comment]
Use alsa, e.g. `amixer sset Master toggle`

**Comment by raveit65** [comment]
This works out of box, switching sound on/off:
```
[rave@mother ~]$ amixer sset Master toggle
Simple mixer control 'Master',0
  Capabilities: pvolume pswitch pswitch-joined
  Playback channels: Front Left - Front Right - Rear Left - Rear Right - Front Center - Woofer
  Limits: Playback 0 - 65536
  Mono:
  Front Left: Playback 29887 [46%] [off]
  Front Right: Playback 29887 [46%] [off]
  Rear Left: Playback 29039 [44%] [off]
  Rear Right: Playback 29039 [44%] [off]
  Front Center: Playback 30762 [47%] [off]
  Woofer: Playback 30762 [47%] [off]
[rave@mother ~]$ amixer sset Master toggle
Simple mixer control 'Master',0
  Capabilities: pvolume pswitch pswitch-joined
  Playback channels: Front Left - Front Right - Rear Left - Rear Right - Front Center - Woofer
  Limits: Playback 0 - 65536
  Mono:
  Front Left: Playback 29887 [46%] [on]
  Front Right: Playback 29887 [46%] [on]
  Rear Left: Playback 29039 [44%] [on]
  Rear Right: Playback 29039 [44%] [on]
  Front Center: Playback 30762 [47%] [on]
  Woofer: Playback 30762 [47%] [on]
```

**Comment by hasufell** [comment]
See the initial post in this thread. You also have to select a card to reproduce the issue.

**Comment by raveit65** [comment]
All i know is that i can mute/unmute with command line or keyboard, but not if i use checkbuttons from popup-window or popup-menu.
Here i can only mute, but not unmute.
So it is reproducible for me.
Please post command what i should do?
Or do you think this isn't a issue?
If yes, why this report is open?

**Comment by raveit65** [comment]
this is my device
```
[rave@mother ~]$ dmesg |grep audio
[    4.215977] snd_hda_intel 0000:0a:00.1: Handle vga_switcheroo audio client
[    4.247019] snd_hda_codec_realtek hdaudioC1D0: autoconfig for ALC1220: line_outs=3 (0x14/0x15/0x16/0x0/0x0) type:line
[    4.247022] snd_hda_codec_realtek hdaudioC1D0:    speaker_outs=0 (0x0/0x0/0x0/0x0/0x0)
[    4.247023] snd_hda_codec_realtek hdaudioC1D0:    hp_outs=1 (0x1b/0x0/0x0/0x0/0x0)
[    4.247024] snd_hda_codec_realtek hdaudioC1D0:    mono: mono_out=0x0
[    4.247026] snd_hda_codec_realtek hdaudioC1D0:    dig-out=0x1e/0x0
[    4.247026] snd_hda_codec_realtek hdaudioC1D0:    inputs:
[    4.247028] snd_hda_codec_realtek hdaudioC1D0:      Front Mic=0x19
[    4.247030] snd_hda_codec_realtek hdaudioC1D0:      Rear Mic=0x18
[    4.247031] snd_hda_codec_realtek hdaudioC1D0:      Line=0x1a
```

**Comment by hasufell** [comment]
Please read the first post in this issue. It describes the problem perfectly and it is reproducible with **all** alsa mixers I know of. Try it with `xfce4-mixer`, try it with `amixer` (see the manpage to set a card for the command). You will see the same bug.

There are also referenced bug reports in the first post which give further information on the problem. There is only one solution: implementing a pulseaudio backend.

**Comment by raveit65** [comment]
Yes, i can reproduce it with using xfce-mixer or any other as mixer device for pnmixer.
I wrote it already in my first post.
Ok, that means switching to pnmixer for Mate desktop for fedora 28 in near future is useless. and i don't need to maintain it in fedora.
Thanks for the info

**Comment by hasufell** [comment]
> Yes, i can reproduce it with using xfce-mixer or any other as mixer device for pnmixer.

I'm not sure what that means. I think you misunderstand how pnmixer works. Pnmixer doesn't use any "backend" except **alsa**. The configurable mixers are just arbitrary commands that are run when you click on the Mixer button for example. They have nothing to do with the audio code.

And I am pretty sure you haven't started `xfce4-mixer` and tried to mute/unmute.

> why not simple using 'amixer sset Master toggle' for the checkbutton

You haven't read the initial post and my follow-up explanations carefully. `amixer` has **the same problem** if you choose a **card**. Refer to the manpage on how to do that.

**Comment by raveit65** [comment]
Common, i know how pnmixer works.
It is base on alsa, but you can select several mixer-frontends, ie. pavucontrol, mate-volume-control an old xfce4-mixer in pnmixer preferences.

> And I am pretty sure you haven't started xfce4-mixer and tried to mute/unmute.
Why do you say that?...,I think you didn't understand what wrote in my first post.
.,.... sigh.
But Ok, i learned that pnmixer isn't a solution for MATE desktop in fedora.
Thanks.

**Comment by hasufell** [comment]
> Why do you say that?...,I think you didn't understand what wrote in my first post.

You were speaking about mixer backends (but there are none). Then you talked about a hotkey combination, where no one knows what it does.

**all alsa mixers have this problem**. This is not pnmixer specific. This is a problem between alsa and pulseaudio communication.

If you have an application where that works, then it doesn't use alsa.
